### PR TITLE
fix gated build failures due to incompatibility with new flake8 and old flake8-bugbear and old flake8-logging-format

### PR DESCRIPTION
**Pull request by imatiach-msft**
The new version of flake8 package is incompatible with the old versions of flake8-bugbear 21.11.29 and old flake8-logging-format 0.6.0.  See related issue:
https://github.com/globality-corp/flake8-logging-format/issues/35

The fix here is to remove the pin on these packages so that we always get the latest versions.  

Specifically for bugbear the error shown is:
```
(flake) PS C:\interpret-community> flake8 --max-line-length=119
Traceback (most recent call last):
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\ilmat\Miniconda3\envs\flake\Scripts\flake8.exe\__main__.py", line 7, in <module>
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\main\cli.py", line 22, in main
    app.run(argv)
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\main\application.py", line 336, in run
    self._run(argv)
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\main\application.py", line 325, in _run
    self.run_checks()
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\main\application.py", line 229, in run_checks
    self.file_checker_manager.run()
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\checker.py", line 252, in run
    self.run_serial()
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\checker.py", line 237, in run_serial
    checker.run_checks()
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\checker.py", line 531, in run_checks
    self.run_ast_checks()
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\flake8\checker.py", line 435, in run_ast_checks
    for (line_number, offset, text, _) in runner:
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\bugbear.py", line 61, in run
    if self.should_warn(e.message[:4]):
  File "C:\Users\ilmat\Miniconda3\envs\flake\lib\site-packages\bugbear.py", line 130, in should_warn
    if code[:i] in self.options.select:
TypeError: argument of type 'NoneType' is not iterable
```

For flake8-logging-format the error is:
```
(flake) PS C:\interpret-community> flake8 --max-line-length=119
There was a critical error during execution of Flake8:
plugin code for `flake8-logging-format[logging-format]` does not match ^[A-Z]{1,3}[0-9]{0,3}$
```

### CODE REVIEW

File: python/interpret_community/tabular_explainer.py [hunk 1]
```diff
@@ -189,13 +189,15 @@ def __init__(self, model, initialization_examples, explain_subset=None, features
                         allow_all_transformations=allow_all_transformations,
                         **kwargs)
                 self._method = self.explainer._method
-                self._logger.info('Initialized valid explainer {} with args {}'.format(self.explainer, kwargs))
+                exp_init = 'Initialized valid explainer {} with args {}'.format(self.explainer, kwargs)
```
Comment (gaugup) [inline python/interpret_community/tabular_explainer.py:192]: Can we rename to exp_init_message?

File: python/interpret_community/tabular_explainer.py [hunk 2]
```diff
@@ -189,13 +189,15 @@ def __init__(self, model, initialization_examples, explain_subset=None, features
...
+                self._logger.info(exp_init)
                 is_valid = True
                 break
             except Exception as ex:
                 last_exception = ex
-                self._logger.info('Failed to initialize explainer {} due to error: {}'
-                                  .format(uninitialized_explainer, ex))
+                exp_init_failed = ('Failed to initialize explainer {} due to error: {}'
```
Comment (gaugup) [inline python/interpret_community/tabular_explainer.py:198]: Can we rename to exp_init_failed_message?

File: tests/common_tabular_tests.py [hunk 3]
```diff
@@ -200,7 +200,7 @@ def _verify_explain_model_local_common(self, model, x_train, x_test, y_train, y_
         assert not explanation.is_raw
         # Validate data has global info
         global_data = explanation.data(key=-1)
-        assert(InterpretData.OVERALL in global_data)
+        assert (InterpretData.OVERALL in global_data)
```
Comment (gaugup) [inline tests/common_tabular_tests.py:203]: Do we need the braces around the assert condition? Same comment for all the assert statements in this review.

File: tests/test_explain_model.py [hunk 4]
```diff
@@ -386,7 +386,8 @@ def _validate_binary_explain_model(self, tabular_explainer, create_model, test_n
         # Create tabular explainer
         exp = tabular_explainer(model, x_train, features=X.columns.values,
                                 classes=classes)
-        test_logger.info('Running explain global for {}'.format(test_name))
+        exp_glob_info = 'Running explain global for {}'.format(test_name)
```
Comment (gaugup) [inline tests/test_explain_model.py:389]: Can we rename to exp_glob_info_message?

File: tests/test_validate_explanations.py [hunk 5]
```diff
@@ -174,16 +174,18 @@ def tabular_explainer_imp(model, x_train, x_test, allow_eval_sampling=True, use_
 
 def validate_correlation(true_order, validate_order, threshold, top_values=10):
     computed_ndcg = ndcg(true_order, validate_order, top_values)
-    test_logger.info("ndcg: " + str(computed_ndcg))
-    assert(computed_ndcg > threshold)
+    ndcg_info = "ndcg: " + str(computed_ndcg)
```
Comment (gaugup) [inline tests/test_validate_explanations.py:177]: Can we rename to ndcg_info_message?

File: tests/test_validate_explanations.py [hunk 6]
```diff
@@ -174,16 +174,18 @@ def tabular_explainer_imp(model, x_train, x_test, allow_eval_sampling=True, use_
...
+    assert (computed_ndcg > threshold)
 
 
 def validate_spearman_correlation(overall_imp, shap_overall_imp, threshold):
     # Calculate the spearman rank-order correlation
     rho, p_val = stats.spearmanr(overall_imp, shap_overall_imp)
     # Validate that the coefficients from the linear model are highly correlated with the results from shap
-    test_logger.info("Calculated spearman correlation coefficient rho: " + str(rho) + " and p_val: " + str(p_val))
-    assert(rho > threshold)
+    spearman_info = "Calculated spearman correlation coefficient rho: " + str(rho) + " and p_val: " + str(p_val)
```
Comment (gaugup) [inline tests/test_validate_explanations.py:186]: Can we rename to spearman_info_message?

**Review by gaugup — CHANGES_REQUESTED** [review]

File: tests/common_tabular_tests.py (hunk 3 above)
Comment (imatiach-msft) [reply tests/common_tabular_tests.py:203]: done, removed

File: python/interpret_community/tabular_explainer.py (hunk 1 above)
Comment (imatiach-msft) [reply python/interpret_community/tabular_explainer.py:192]: done

File: python/interpret_community/tabular_explainer.py (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/tabular_explainer.py:198]: done

File: tests/test_explain_model.py (hunk 4 above)
Comment (imatiach-msft) [reply tests/test_explain_model.py:389]: done!

File: tests/test_validate_explanations.py (hunk 5 above)
Comment (imatiach-msft) [reply tests/test_validate_explanations.py:177]: done!

**Review by gaugup — APPROVED** [review]

**Pull request merged**